sn_record_store/src/lib.rs

```diff
@@ -216,7 +216,11 @@ impl RecordStore for DiskBackedRecordStore {
         let num_records = self.records.len();
         if num_records >= self.config.max_records {
             warn!("Record not stored. Maximum number of records reached. Current num_records: {num_records}");
-            return Err(Error::MaxRecords);
+            // To avoid returning error causing libp2p holding un-necessary copy in cache,
+            // to achieve specified `quorumn::ALL` successes,
+            // which may result in OOM(Out Of Memory) issue under certain network circumstance.
+            // Here returning `OK` instead of `Error::MaxRecords`.
+            return Ok(());
```

Comment:
Hmm, this feels like it breaks the underlying kad... we have not stored the data...

The copies in cache are held because of this no? Until we have a node with sufficient capacity? So really we'd want that extra capcity, or to prune... (And in the end put pressure to pay nodes more to get that space). 

I think for now this may be an expected issue if we're not growing the network

Reply:
I'd even say stay, keep cache (which I assume should be disbersed upon new members), and make sure we're updating store cost... If we can assess this cache size, we miiight have a real nice level data velocity too perhaps? (If it's cached until written... higher cache w/o being full will mean we have a data bottleneck and so should also increase storecost?